src/v/redpanda/admin_server.cc

```diff
@@ -4581,3 +4601,130 @@ admin_server::restart_service_handler(std::unique_ptr<ss::http::request> req) {
+        return "initialized";
+    case debug_bundle_status::in_progress:
+        return "in-progress";
+    case debug_bundle_status::finished:
+        return "finished";
+    }
+    return "unknown";
+}
+
+template<>
+constexpr std::optional<debug_bundle_status>
+from_string_view<debug_bundle_status>(std::string_view sv) {
+    return string_switch<std::optional<debug_bundle_status>>(sv)
+      .match(
+        to_string_view(debug_bundle_status::initialized),
+        debug_bundle_status::initialized)
+      .match(
+        to_string_view(debug_bundle_status::in_progress),
+        debug_bundle_status::in_progress)
+      .match(
+        to_string_view(debug_bundle_status::finished),
+        debug_bundle_status::finished)
+      .default_match(std::nullopt);
+}
+
+struct process_consumer {
+    using consumption_result_type =
+      typename ss::input_stream<char>::consumption_result_type;
+    using stop_consuming_type =
+      typename consumption_result_type::stop_consuming_type;
+    using tmp_buf = stop_consuming_type::tmp_buf;
+
+    process_consumer(debug_bundle& bundle)
+      : bundle{bundle} {}
+
+    ss::future<consumption_result_type> operator()(tmp_buf buf) {
+        std::string str_buf{buf.begin(), buf.end()}, line;
+        vlog(logger.info, "Consumer read {}", str_buf);
+        std::stringstream ss(str_buf);
+        while (!ss.eof()) {
+            std::getline(ss, line);
+            if (line.starts_with("Debug bundle saved to")) {
+                vlog(logger.info, "Stop condition reached on line {}", line);
+                bundle.status = debug_bundle_status::finished;
+                return make_ready_future<consumption_result_type>(
+                  stop_consuming_type({}));
+            }
+        }
+        return make_ready_future<consumption_result_type>(
+          ss::continue_consuming{});
+    }
+    debug_bundle& bundle;
+};
+
+// This code was modified from seastar's spawn_test_echo in
+// https://github.com/scylladb/seastar/blob/master/tests/unit/spawn_test.cc#L64
+void admin_server::make_debug_bundle_in_background() {
+// RPK_BINARY_PATH is set in the task rule "rp:build" and the value is set as a
+// compile time macro in src/v/redpanda/CMakeLists.txt
+#ifdef RPK_BINARY_PATH
+    const char* rpk_cmd = RPK_BINARY_PATH;
+#else
+    const char* rpk_cmd = "/usr/bin/rpk";
+#endif
+
+    vlog(logger.trace, "RPK PATH {}", rpk_cmd);
+    ssx::background
+      = ss::experimental::spawn_process(
+          rpk_cmd,
+          {.argv
+           = {rpk_cmd, "debug", "bundle", "--output", _bundle.filename.c_str()}})
+          .then([this](auto process) {
+              auto stdout = process.stdout();
+              _bundle.status = debug_bundle_status::in_progress;
+              return do_with(
+                std::move(process),
+                std::move(stdout),
+                [this](auto& p, auto& stdout) {
+                    return stdout.consume(process_consumer(_bundle))
```

Comment:
is the only interface available for reading from stdout the consume(consumer()) interface, or is there also an interface where you can stdout.read() and get some bytes back?

Reply:
I was not able to get output from `stdout.read()` so I switched to this implementation so I could focus on running the bundle.